pkg/perf/perf.go

```diff
@@ -55,6 +58,12 @@ var (
 	ErrNoSymbolFound      = errors.New("no symbol found")
 )
 
+var perfFileFmts = []string{
+	"/proc/%d/root/tmp/perf-%d.map",
+	"/proc/%d/root/tmp/jit-%d.dump",
+	"/proc/%d/cwd/jit-%d.dump",
```

Comment:
LLVM supports an environment variable `JITDUMPDIR`

https://github.com/llvm/llvm-project/blob/bb666c69300c962c6a259c9d8e504601e557bc30/llvm/lib/ExecutionEngine/PerfJITEvents/PerfJITEventListener.cpp#L304

and otherwise will be `.debug/jit/llvm-IR-jit-20201109-09a2b8/jit-2397822.dump` in either `$HOME` or `cwd`.

Reply:
`perf inject --jit` get's the filename from the corresponding perf event in the main file.

Reply:
Correct (https://github.com/parca-dev/parca-demo/pull/24#issuecomment-1331627461), I will add tackle that in the next PR

> `perf inject --jit` get's the filename from the corresponding perf event in the main file.

Do you know the details about how the filename ends up in the `perf.data` file? I had in mind to list all files under `.debug/jit/` ordered by mtime

Reply:
From what I can tell you would need to parse the perf events and look for the MMAP2 and MMAP records (see tools/perf/perf-inject.c) the events are described here https://github.com/torvalds/linux/blob/97ee9d1c16963375eefdf964c429897d27e28956/include/uapi/linux/perf_event.h#L831

And then you can check that it ends in `jit-$PID.dump`

Reply:
Thank you, I think I see how it works on a high level, the agent does not record these events at this time and the target process can start before the agent, so it does not seem we will be able to use that

Reply:
Actually it looks like you could simple scan /proc/.../maps according to https://github.com/llvm/llvm-project/blob/bb666c69300c962c6a259c9d8e504601e557bc30/llvm/lib/ExecutionEngine/PerfJITEvents/PerfJITEventListener.cpp#L341

Reply:
```
➜  ~ cat /proc/34437/maps | grep -G 'jit-[[:digit:]]*.dump' -
7fe2b5123000-7fe2b5124000 r-xp 00000000 103:03 21235307                  /home/vchuravy/.debug/jit/llvm-IR-jit-20221203-36c97d/jit-34437.dump
```

Reply:
Nice! Thank you so much for the research!